src/main/java/com/google/ehub/data/CommentDataManager.java

```diff
@@ -0,0 +1,43 @@
+package com.google.ehub.data;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import java.util.*;
+
+public class CommentDataManager{
+
+  public void addItemComment(long itemId, String message, long timestamp){
+    Entity commentEntity = new Entity("ItemPageComments");
+    commentEntity.setProperty("itemId", itemId);
+    commentEntity.setProperty("message", message);
+    commentEntity.setProperty("timestamp",timestamp);
+
+    DatastoreService commentDatastoreInitialization = DatastoreServiceFactory.getDatastoreService();
+    commentDatastoreInitialization.put(commentEntity);
+
+  }
+
+
+  public ArrayList<CommentData> retrieveItemComment(long itemId){
```

Comment:
Prefer List over ArrayList. Its better to refer to objects by their interfaces because your program is a lot more flexible then. You can switch the implementation without having to change the API. Using List is acceptable here because it still has the same functionality that you want from the ArrayList.

Also, its better to name this method `retrieveItemComments` or simply `retrieveComments` (plural) since you are returning a list of comments. 

```suggestion
  public List<CommentData> retrieveItemComments(long itemId) {
```